### PR TITLE
Compile nginx from sources 

### PR DESCRIPTION
**Pull request by rmars**
In order to use the branch https://github.com/ghedo/headers-more-nginx-module/tree/wildcard_in to use wildcards to clear input headers.

### CODE REVIEW

File: docker/nginx/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Docker image that compiles NGINX from source
+
+The nginx Headers More module doesn't currently accept wildcard matches
+for clearing input headers (it's on a branch somehwere). So if we want to use this
+capability we need to compile nginx from source and add this module.
+
+We're building a custom docker image from a branch of the [Headers More](https://github.com/openresty/headers-more-nginx-module) module. This dockerfile installs the tools necessary
+to compile nginx and the headers more module from source, then copies in our
+custom nginx config.
```
Comment (klingerf) [inline docker/nginx/README.md]: I think you can remove the ", then copies in our custom nginx config." part here -- that's still being mounted from the kubernetes configmap.

File: docker/nginx/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+This image is pre-built and available on docker hub as[buoyantio/nginx:1.10.2](https://hub.docker.com/r/buoyantio/nginx/tags/),
+so there should be no need to rebuild it.
+If you would still like to build this, get a copy of the
+[wildcard_in](https://github.com/ghedo/headers-more-nginx-module/tree/wildcard_in) branch, and adjust this line in the dockerfile to get the module files
+from wherever you've saved the branch.
+
+```
+ADD headers-more-nginx-module /headers-more-nginx-module
+```
```
Comment (klingerf) [inline docker/nginx/README.md]: I might simplify this a bit.  Something like:

If you would still like to build this image, you'll first need to locally clone and check out a custom branch of the `headers-more-nginx-module` repo, as follows:

```
$ git clone https://github.com/ghedo/headers-more-nginx-module.git
$ cd headers-more-nginx-module
$ git checkout wildcard_in
$ cd ..
```

---

By being more direct about getting the clone locally, you can avoid having people modify the dockerfile.

File: docker/nginx/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+We're building a custom docker image from a branch of the [Headers More](https://github.com/openresty/headers-more-nginx-module) module. This dockerfile installs the tools necessary
+to compile nginx and the headers more module from source, then copies in our
+custom nginx config.
+
+This image is pre-built and available on docker hub as[buoyantio/nginx:1.10.2](https://hub.docker.com/r/buoyantio/nginx/tags/),
+so there should be no need to rebuild it.
+If you would still like to build this, get a copy of the
+[wildcard_in](https://github.com/ghedo/headers-more-nginx-module/tree/wildcard_in) branch, and adjust this line in the dockerfile to get the module files
+from wherever you've saved the branch.
```
Comment (esbie) [inline docker/nginx/README.md]: this part was confusing to me.... I would make these steps more declarative

```
git clone -b wildcard_in git@github.com:ghedo/headers-more-nginx-module.git
docker build --tag="buoyantio/nginx:1.10.2" -f Dockerfile .
```

File: docker/nginx/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+## Docker image that compiles NGINX from source
+
+The nginx Headers More module doesn't currently accept wildcard matches
+for clearing input headers (it's on a branch somehwere). So if we want to use this
```
Comment (esbie) [inline docker/nginx/README.md]: it's on a branch somewhere????? either link to it or don't include the parens. (also somewhere typo)

File: docker/nginx/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+RUN apt-get -y install libpcre3 libpcre3-dev zlib1g-dev libssl-dev
+
+ADD headers-more-nginx-module /headers-more-nginx-module
+
+RUN wget http://nginx.org/download/nginx-1.10.2.tar.gz
+RUN tar zxf nginx-1.10.2.tar.gz && cd nginx-1.10.2 && ./configure --prefix=/etc/nginx --sbin-path=/usr/sbin/nginx --conf-path=/etc/nginx/nginx.conf --error-log-path=/var/log/nginx/error.log --http-log-path=/var/log/nginx/access.log --pid-path=/var/run/nginx.pid --lock-path=/var/run/nginx.lock --add-module=/headers-more-nginx-module && make && sudo make install
+
+EXPOSE 80 443
+
+CMD ["nginx", "-g", "daemon off;"]
```
Comment (esbie) [inline docker/nginx/Dockerfile]: nit: eof newline

**Pull request merged**